src/Data/Either/Open.purs

```diff
@@ -0,0 +1,146 @@
+module Data.Either.Open where
```

Comment:
I would replace `Data.Either.Nested` with this. There is not room for both of them, and this is a much better approach which we can also replicate to `Date.Tuple` (but with `Unit` in place of `Void`).